aten/src/THC/generic/THCTensorMathMagma.cu

```diff
@@ -235,7 +235,12 @@ THC_API void THCTensor_(syev)(THCState *state, THCTensor *re_, THCTensor *rv_, T
     else if (info < 0)
       THError("MAGMA syev : Argument %d : illegal value", -info);
   }
-  THCTensor_(freeCopyTo)(state, input, rv_);
+  if (jobzs[0] != 'N') {
+    THCTensor_(freeCopyTo)(state, input, rv_);
+  } else  {
+    // If eigenvector is not needed, fill the result with zeros.
```

Comment:
Is this not the other way round, based on code above? `jobzs[0] == 'N'` gives `MagmaNoVec`. I think it would be nice to have tests too.